src/plugins/explore/public/components/top_nav/top_nav.tsx

```diff
@@ -171,27 +182,59 @@ export const TopNav = ({ setHeaderActionMenu = () => {}, savedExplore }: TopNavP
     [queryString, data.dataViews, dispatch, services, clearEditors]
   );
 
+  // Custom onChange handler to track date range changes in Redux (mirrors SearchBar behavior)
+  const handleQueryChange = useCallback(
+    (queryAndDateRange: { dateRange: any; query?: Query }) => {
+      if (queryAndDateRange.dateRange) {
+        dispatch(setDateRange(queryAndDateRange.dateRange));
+      }
+    },
+    [dispatch]
+  );
+
+  const handleQuerySubmit = useCallback(() => {
+    dispatch(onEditorRunActionCreator(services, editorText));
+    dispatch(setQueryExecutionButtonStatus('REFRESH'));
+  }, [dispatch, services, editorText]);
+
+  const handleCustomButtonClick = useCallback(
+    (event: React.MouseEvent<HTMLButtonElement>) => {
+      handleQuerySubmit();
+    },
+    [handleQuerySubmit]
+  );
+
+  const customSubmitButton = useMemo(
+    () => <QueryExecutionButton editorText={editorText} onClick={handleCustomButtonClick} />,
+    [editorText, handleCustomButtonClick]
+  );
+
   return (
     <TopNavMenu
       appName={PLUGIN_ID}
       config={topNavLinks}
       data={data}
-      showSearchBar={false}
+      showSearchBar={TopNavMenuItemRenderType.IN_PLACE}
       showDatePicker={showDatePicker && TopNavMenuItemRenderType.IN_PORTAL}
       showSaveQuery={false}
       showDatasetSelect={true}
       datasetSelectProps={{
         onSelect: handleDatasetSelect,
       }}
-      useDefaultBehaviors
+      useDefaultBehaviors={false}
       setMenuMountPoint={setHeaderActionMenu}
       indexPatterns={dataset ? [dataset] : datasets}
       savedQueryId={undefined}
       onSavedQueryIdChange={() => {}}
+      onQuerySubmit={handleQuerySubmit}
+      onQueryChange={handleQueryChange}
+      customSubmitButton={customSubmitButton}
       groupActions={true}
       screenTitle={screenTitle}
       queryStatus={queryStatus}
-      showQueryBar={false}
+      showQueryBar={true}
+      showQueryInput={false}
+      showFilterBar={false}
```

Comment:
how is the datepicker getting rendered? I see that you are passing a customSubmitZButton, but not a datepicker?